event_reporter/reporter/application_errors_parser.go

```diff
@@ -52,10 +48,13 @@ func parseApplicationSyncResultErrorsFromConditions(status appv1.ApplicationStat
 			resourcesSyncErrors := parseAggregativeResourcesSyncErrors(status.OperationState.SyncResult.Resources)
 
 			errs = append(errs, resourcesSyncErrors...)
-		} else {
+			continue
+		}
+
+		if cnd.IsError() || cnd.IsWarning() {
```

Comment:
maybe better to check that conditional level is not empty, so you will have single place